src/cache/redis/redis.cache.service.ts

```diff
@@ -0,0 +1,36 @@
+import type { RedisClient } from './redis.cache.module';
+
+import { Inject, Injectable } from '@nestjs/common';
+import { REDIS_CLIENT } from './redis.cache.module';
+
+@Injectable()
+export class CacheService {
+  constructor(@Inject(REDIS_CLIENT) private readonly _redis: RedisClient) {}
+
+  /**
+   * Gets the Redis client.
+   * @returns {RedisClient} The Redis client.
+   */
+  get client(): RedisClient {
+    return this._redis;
+  }
+
+  /**
+   * Gets the value at the specified key.
+   * @param {string} key - The key to get the value from.
+   * @returns {Promise<string | null>} The result of the get operation.
+   */
+  async get(key: string): Promise<string | null> {
+    return this._redis.get(key);
+  }
+
+  /**
+   * Sets the value at the specified key.
+   * @param {string} key - The key to set the value at.
+   * @param {string | number} value - The value to set at the specified key.
+   * @returns {Promise<void>} The result of the set operation.
+   */
+  async set(key: string, value: string | number): Promise<void> {
```

Comment:
```suggestion
  async set<T = unknown>(key: SomeLimited, value: T): Promise<void> {
```
Should we limit somehow the keys that can be used to avoid misuse of the cache since it's a payed service?
Also I wonder if this should be a generic function. Why would I want it to be just a string or a number? Is it because in redis I can only save those types? But then shouldn't be this service's responsibility to make sure it is saved in the right format? Maybe stringifying the object so that we ensure we can then parse it in the same way?

Reply:
I tend to think this happens not on this layer, but on the consumer of the cache.

Reply:
Makes sense.
I do agree that it's partially the consumer's responsibility to do that.
But on the other hand I would like to ensure a certain degree of consistency of the data that I store in my cache. Redis seems to be supporting natively [different data types](https://redis.io/docs/latest/develop/data-types/). Since we are not exposing the whole API then shouldn't be responsible of at least trying to take advantage of some of that information?

Reply:
I responded before reading your latest answer, sorry! heh
To your point, yes, we could explore how data is mapped and expose a better API, for sure.
Let's keep that in mind as a possible improvement!